server/implementation/src/main/java/io/smallrye/graphql/scalar/custom/CustomFloatScalar.java

```diff
@@ -0,0 +1,48 @@
+package io.smallrye.graphql.scalar.custom;
+
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+
+import jakarta.json.JsonValue.ValueType;
+import jakarta.json.bind.serializer.DeserializationContext;
+import jakarta.json.bind.serializer.JsonbDeserializer;
+import jakarta.json.bind.serializer.JsonbSerializer;
+import jakarta.json.bind.serializer.SerializationContext;
+import jakarta.json.stream.JsonGenerator;
+import jakarta.json.stream.JsonParser;
+
+import io.smallrye.graphql.spi.ClassloadingService;
+
+public interface CustomFloatScalar {
+    // Note: using lambdas for the SERIALIZER/DESERIALIZER instances doesn't work because it
+    // hides the parameterized type from Jsonb.
+
+    JsonbSerializer<CustomFloatScalar> SERIALIZER = new JsonbSerializer<>() {
+        @Override
+        public void serialize(CustomFloatScalar customFloatScalar, JsonGenerator jsonGenerator,
+                SerializationContext serializationContext) {
+            jsonGenerator.write(customFloatScalar.floatValue());
+        }
+    };
+
+    JsonbDeserializer<CustomFloatScalar> DESERIALIZER = new JsonbDeserializer<>() {
+        @Override
+        public CustomFloatScalar deserialize(JsonParser jsonParser,
+                DeserializationContext deserializationContext, Type type) {
+            ClassloadingService classloadingService = ClassloadingService.get();
+            try {
+                if (jsonParser.getValue().getValueType() == ValueType.NULL) {
+                    return null;
+                } else {
+                    return (CustomFloatScalar) classloadingService.loadClass(type.getTypeName())
+                            .getConstructor(BigDecimal.class)
+                            .newInstance(jsonParser.getBigDecimal());
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    };
+
+    BigDecimal floatValue();
```

Comment:
Hmmm... Would it be more understandable to call this method `serialize`, or `valueForSerialization`, or something, because it is used for serializing?

Reply:
I think thats a reasonable name for this accessor method.